File: 229_Majority_Element_II.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int frequency = (int) std::floor(nums.size() / 3);
        map<int, int> k;
        vector<int> r; 
        for(int& c : nums){
            if(k.find(c) != k.end()){
                k[c]++;  
            }else{
                k[c] = 1; 
            }
        }
        
        for(auto const&[key, val]: k)
        {
            if(val > frequency) r.push_back(key); 
        }
        return r; 
    }
};

File: 229_Majority_Element_II.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        List<Integer> r = new ArrayList<Integer>(); 
        HashMap<Integer, Integer> k = new HashMap<Integer, Integer>(); 
        int frequency = nums.length / 3; 
        for(int i : nums)
        {
            if(k.containsKey(i)){
                k.put(i, k.get(i) + 1); 
            }
            else{
                k.put(i, 1); 
            }
            
            if((k.get(i) > frequency) && (!r.contains(i)))
            {
                r.add(i); 
            }
        }
        
        return r; 
    }
}